DMCompiler/DM/DMProc.cs

```diff
@@ -111,6 +123,9 @@ private class DMProcScope {
         public void AddLocalVariable(string name, DreamPath? type) {
             int localVarId = _localVariableIdCounter++;
 
+            if (_scopes.Peek().LocalVariables.ContainsKey(name)) {
+                throw new CompileErrorException($"attempt to redefine variable {name}");
+            }
             _scopes.Peek().LocalVariables.Add(name, new DMLocalVariable(localVarId, type));
```

Comment:
No need to `Peek()` twice
```suggestion
            DMProcScope scope = _scopes.Peek();
            if (scope.LocalVariables.ContainsKey(name)) {
                throw new CompileErrorException($"attempt to redefine variable {name}");
            }
            scope.LocalVariables.Add(name, new DMLocalVariable(localVarId, type));
```